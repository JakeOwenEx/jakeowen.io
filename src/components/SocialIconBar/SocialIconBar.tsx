import * as React from "react";
import GitHubLogo from "../../assets/svgs/socialIcons/github.svg";
import MediumLogo from "../../assets/svgs/socialIcons/medium.svg";
import LinkedInLogo from "../../assets/svgs/socialIcons/linkedin.svg";

interface Icon {
  alt: string;
  src: any;
}

const SocialIconImage: React.FC<Icon> = (props) => {
  return (
    <div>
      <a target="_blank">
        <img
          className={`social-icon ${props.alt}`}
          src={props.src}
          alt={props.alt}
        />
      </a>
    </div>
  );
};

const SocialIconBar: React.FC = () => {
  return (
    <div className="social-icon-bar">
      <SocialIconImage src={GitHubLogo} alt="GitHubLogo" />
      <SocialIconImage src={MediumLogo} alt="MediumLogo" />
      <SocialIconImage src={LinkedInLogo} alt="LinkedInLogo" />
      <div className="social-icon-grey-bar" />
    </div>
  );
};

export default SocialIconBar;
