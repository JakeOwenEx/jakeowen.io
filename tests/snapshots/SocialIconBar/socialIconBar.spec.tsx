// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
import * as React from "react";
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
import { create } from "react-test-renderer";
import SocialIconBar from "../../../src/components/SocialIconBar/SocialIconBar";

describe("Social bar Snapshot ", () => {
  it("Social Icon Bar", () => {
    expect(create(<SocialIconBar />)).toMatchSnapshot();
  });
});
