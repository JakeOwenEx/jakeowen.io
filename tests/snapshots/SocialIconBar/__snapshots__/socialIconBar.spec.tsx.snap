// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social bar Snapshot  Social Icon Bar 1`] = `
<div
  className="social-icon-bar"
>
  <div>
    <a
      target="_blank"
    >
      <img
        alt="GitHubLogo"
        className="social-icon GitHubLogo"
      />
    </a>
  </div>
  <div>
    <a
      target="_blank"
    >
      <img
        alt="MediumLogo"
        className="social-icon MediumLogo"
      />
    </a>
  </div>
  <div>
    <a
      target="_blank"
    >
      <img
        alt="LinkedInLogo"
        className="social-icon LinkedInLogo"
      />
    </a>
  </div>
  <div
    className="social-icon-grey-bar"
  />
</div>
`;
